### PR TITLE
drop(grouping): Drop split hashes

### PR DESCRIPTION
**Pull request by armenzg**
This is part of undoing hierarchical grouping code.

### CODE REVIEW

File: src/sentry/models/grouphash.py [hunk 1]
```diff
@@ -26,9 +26,6 @@ class State:
         UNLOCKED = None
         LOCKED_IN_MIGRATION = 1
 
-        # This hierarchical grouphash should be ignored/skipped for finding the group.
-        SPLIT = 2
```
Comment (armenzg) [inline src/sentry/models/grouphash.py:30]: This got added in https://github.com/getsentry/sentry/pull/25481

File: src/sentry/api/helpers/group_index/delete.py [hunk 2]
```diff
@@ -49,13 +49,7 @@ def delete_group_list(
     # Tell seer to delete grouping records for these groups
     call_delete_seer_grouping_records_by_hash(group_ids)
 
-    # We do not want to delete split hashes as they are necessary for keeping groups... split.
-    GroupHash.objects.filter(
-        project_id=project.id, group__id__in=group_ids, state=GroupHash.State.SPLIT
-    ).update(group=None)
-    GroupHash.objects.filter(project_id=project.id, group__id__in=group_ids).exclude(
-        state=GroupHash.State.SPLIT
-    ).delete()
```
Comment (armenzg) [inline src/sentry/api/helpers/group_index/delete.py:58]: This got added in #28475

File: src/sentry/models/grouphash.py (hunk 1 above)
Comment (snigdhas) [reply src/sentry/models/grouphash.py:30]: Are there any rows still using this value? I've usually seen the constant kept around with a deprecated comment for compatibilitiy.

Comment (armenzg) [reply src/sentry/models/grouphash.py:30]: This query returns that 1133 rows are in the LOCKED_IN_MIGRATION and none on SPLIT.
```
select
  state,
  count(*)
from
  getsentry.sentry_grouphash as gh
where
  gh.state is not null
group by gh.state;
```

Comment (snigdhas) [reply src/sentry/models/grouphash.py:30]: thanks for checking -- if this isn't incompatible for self hosted, then lgtm!

**Review by snigdhas — APPROVED** [review]

**Pull request merged**